### DISCUSSION TITLE
Inline hyperlinks

### DISCUSSION THREAD

**Discussion by TomEdwardsEnscape** [AvaloniaUI/Avalonia#8818] [Ideas]
The new text inlines in Avalonia 11 are a great feature. But there is one WPF inline that is notably absent: [Hyperlink](https://docs.microsoft.com/en-us/dotnet/api/system.windows.documents.hyperlink?view=windowsdesktop-6.0). This allows us to insert a clickable text inline that raises events on mouse interaction and can execute an command on click.

I see that Avalonia text inlines do not implement `IInteractive`, so there is currently no easy way to make one respond to mouse input. Are there plans to change this?

**Comment by Gillibald** [comment]
There are plan's to refactor IInputElement so it is no longer a IVisual but no work towards that has been done yet. In the meantime you can restyle a button and add it via InlineUIContainer.

**Reply by TomEdwardsEnscape** [reply]
Thanks! Is this planned for Avalonia 11?

**Comment by HelenaUdd** [comment]
I'm requesting this feature too! Has anything happened in the last year, are hyperlinks implemented?

**Reply by JamesBalisciano** [reply]
Issue #10186 looks to still be open and unassigned. I would also really like to see this.

**Reply by timunie** [reply]
Will not happen before 11.0 release. After that, probably but not 100 % sure as it needs to add more value than just a restyled button. 

Note: We had some community members trying to implement such a control, but it's not straight forward in the X-plat world as you need to handle links different debending on your platform. Linux is different than browser and mobile is another story.

**Reply by TomEdwardsEnscape** [reply]
The value of an inline hyperlink is that it lives within a `TextBox` and takes part in text wrapping, truncation, and so on. A button, even embedded via `InlineUIContainer`, cannot provide this.

**Reply by timunie** [reply]
That's true and I fully agree that this is something we want to have some day, but as it is not a breaking change it can also be added to 11.x

**Reply by robloo** [reply]
@timunie Don't confuse an inline hyperlink with a HyperlinkButton control. They are very different use cases. An inline hyperlink hasn't been attempted yet.

Inline hyperlink is great for documents which is now a big use case I'm Avalonia 11.0 which unlocks a lot of the other flow document functionality.

HyperlinkButton will be completed and I'll do it myself if needed after a few months.

**Reply by timunie** [reply]
@robloo thanks for clarifying this again 👍 I must have overseen the *inline* hint here.

**Comment by Gillibald** [comment]
https://github.com/AvaloniaUI/Avalonia/issues/10186 is the main blocker here
